dev-tools/cloud/terraform/main.tf

```diff
@@ -0,0 +1,45 @@
+terraform {
+  required_version = ">= 0.12.29"
+
+  required_providers {
+    ec = {
+      source  = "elastic/ec"
+      version = "0.5.1"
+    }
+  }
+}
+
+provider "ec" {}
+
+variable "elastic_agent_docker_image" {
+  type        = string
+  description = "Elastic agent docker image with tag."
+}
+
+locals {
+  match           = regex("const DefaultVersion = \"(.*)\"", file("${path.module}/../../../version/version.go"))[0]
+  stack_version   = format("%s-SNAPSHOT", local.match)
+  docker_image_ea = var.elastic_agent_docker_image
+}
+
+resource "ec_deployment" "deployment" {
+  name                   = "example"
```

Comment:
Is there any check or safeguard to avoid creating more than one deployment ? Would terraform fail if that tries to be done since it is using the name resource? 

If it could be created several deployments, I would think adding here some suffix with some random string is needed. Mainly, it would be needed to avoid create another deployment with the same name.

One option it would be to add a new variable that should be filled by the script:
```
  name                   = "example-${var.DEPLOY_ID}"
```

Probably, this would cause some issue with the `cloud-clean` target, since the same value should be used.

Reply:
If this is going to be run in every PR , I guess it should be added some kind of suffix here.